middleware/authenticated.js

```diff
@@ -43,15 +43,18 @@ export default function ({
     res.set('Cache-Control', 'private');
     res.set('Vary', 'Cookie');
   }
+  const {
+    register, from, referrer, language,
+  } = query;
   if (!store.getters.getUserIsRegistered) {
     let redirectPath = '/in/register';
-    const { register, from, referrer } = query;
     if (!process.server) {
       store.commit(USER_SET_AFTER_AUTH_ROUTE, route);
       setRouteToSessionStorage(route);
-      if (register === '1') redirectPath = `${redirectPath}?register=1`;
+      if (register === '1') redirectPath = `${redirectPath}?register=1&language=${language}`;
```

Comment:
Will language be set as `undefined` if it is not set? should we check for `language` exists or default it to `''` ?

Reply:
Added default

Reply:
I think it's better to not import the default locale into this file. Can we not handle language if it is not set in qs, and focus the handling of default locale in the i18n plugin file?

Reply:
Updated